web/src/settings_bots.js

```diff
@@ -418,8 +418,6 @@ export function show_edit_bot_info_modal(user_id, from_user_info_popover) {
 
         function get_options() {
             const user_ids = people.get_active_human_ids();
-            // Remove current owner.
-            user_ids.splice(user_ids.indexOf(owner_id), 1);
             return user_ids.map((user_id) => ({
```

Comment:
@Ddharmani3 I understand what you are trying to fix, but I intentionally wanted to remove the bot owner out of the options since it doesn't make sense to choose the current owner again.

The actual fix is to get `owner_id = bot_data.get(user_id).owner_id;` again in `get_options` here, since it looks like even though the global variable `owner_id` is updated here, its value inside `get_options` doesn't change?

Reply:
See https://chat.zulip.org/#narrow/stream/92-learning/topic/global.20vs.20local.20variable

Reply:
@amanagr, but the owner id updates after clicking save changes but what if the user has changed the owner but again wants to change it back to the original owner while the edit modal is open?

If we still don't want to show the currently selected value in the dropdown, I can do `user_ids.splice(user_ids.indexOf($("#edit_bot_modal .bot_owner_id").val(), 1);` once #24910 gets merged.

Reply:
> but the owner id updates after clicking save changes but what if the user has changed the owner but again wants to change it back to the original owner while the edit modal is open?

That's not an issue here, since we only want to avoid the current owner. If you use `owner_id = bot_data.get(user_id).owner_id;` inside `get_options`, you will immediately realize it.